lib/uwb/UwbMacAddress.cxx

```diff
@@ -85,6 +85,32 @@ UwbMacAddress::Random(UwbMacAddressType type)
     }
 }
 
+/* static */
+std::optional<std::unordered_set<UwbMacAddress>>
```

Comment:
There's no need to differentiate between the two. Basically, it's either "regex matches so return the set of mac addresses" or "regex doesn't match so we can't return any valid mac addresses". So I can definitely remove the `std::optional` and have the caller check `!empty()`.

Reply:
Yep, got it, thanks for the explanation and additional context.

I agree with the usage of `std::optional` here assuming the utility function was rescoped to be specific to `nocli` (I think you did this). The reason being that using the contextual type (`std::optional`) would always be used for the code this function is intending to help. Below is just aside discussion.

If it hadn't been moved and instead was a more general utility/helper function, then I'd say drop `std::optional` because increases the dependency requirements for other callers that might not care about `std::optional`. The principal is essentially to design the function to work with the most specific type and the fewest layers of indirection, letting the callers mutate/wrap/build up the result into whatever they need (eg. `std::optional`, a container, a `std::atomic`, etc.).